core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -113,6 +113,12 @@ public static StructLikeSet toEqualitySet(CloseableIterable<StructLike> eqDelete
     return new PositionStreamDeleteFilter<>(rows, rowToPosition, posDeletes);
   }
 
+  public static <T> CloseableIterable<T> streamingSelector(CloseableIterable<T> rows,
```

Comment:
This name doesn't look quite clear enough. How about `streamingDeletedRowFilter`? I think that's clear.